src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -1681,6 +1681,19 @@ def _openssh_public_key_bytes(self, key):
             )
 
 
+# If the backend has support for Scrypt, dynamically register the backend and
+# implement the interface.
+def _derive_scrypt(self, key_material, salt, length, n, r, p):
```

Comment:
Won't this break if the user uses the OpenSSL backend directly instead of going through default_backend()?